### DISCUSSION TITLE
Select rows from clipboard

### DISCUSSION THREAD

**Discussion by geekscrapy** [saulpw/visidata#835] [Show and tell]
Scenario: You have a basic list of values on the system clipboard that you need to select matching rows for in a column. I use this when I have a list of IPs I need to search for.

To use:
Paste into .visidatarc
Copy a list to a your system clipboard
Use `select-rows-frm-clipboard` when the cursor is over a column with the according data.

Catch: Values in the clipboard must be regex friendly or regex escaped

```
Sheet.addCommand(None, 'select-rows-frm-clipboard', 'sheet.SelectFromClipboard()', 'matches lines from clipboard and selects matching rows in current column')
@Sheet.api
@asyncthread
def SelectFromClipboard(sheet):
  text = clipboard().paste() or vd.error('system clipboard is empty')
  lines = set([ l.strip() for l in text.splitlines() ])
  vd.status(f'searching for {len(lines)} items')
  for line in Progress(lines, gerund='selecting'):
    sheet.selectByIdx(vd.searchRegex(sheet, regex=line, columns="cursorCol"))
```

If someone can point me in the direction to do the equivalent of search exact match instead of `vd.searchRegex`, then I could add this in!

**Comment by saulpw** [comment]
I think in this case you'd use `select(gatherBy())` with a custom function that checks all values at once for the row.  Like in `select-exact-row`.

**Reply by geekscrapy** [reply]
Thanks Saul, I'll give it a go!
